File: leetcode/medium/161-one-edit-distance/161-one-edit-distance.py

class Solution:
    def isOneEditDistance(self, s: str, t: str) -> bool:
        if s == t:
            return False

        if len(t) > len(s):
            s, t = t, s

        for i in range(len(s)):
            if i < len(t) and s[i] != t[i]:
                if (len(s) == len(t)):
                    return s[i + 1:] == t[i + 1:]

                return s[i + 1:] == t[i:]

            if i > len(t):
                return False

        return True

File: leetcode/medium/161-one-edit-distance/161-one-edit-distance.kt


class Solution {
    fun isOneEditDistance(s: String, t: String): Boolean {
        if (s == t) {
            return false
        }

        if (s.length < t.length) {
            return isOneEditDistance(t, s)
        }

        for (i in 0 until s.length) {
            if (i >= t.length) {
                if (i + 1 == s.length) {
                    return true
                }
                return false
            }

            if (s[i] != t[i]) {
                if (s.length == t.length) {
                    return s.substring(i + 1) == t.substring(i + 1)
                } else {
                    return s.substring(i + 1) == t.substring(i)
                }
            }
        }

        return true
    }
}
